### ISSUE TITLE
e._formatValueItem Cannot read property 'find' of undefined

### ISSUE THREAD

**Issue by indraraj26** [eakoriakin/ionic-selectable#139]
I dont get any error in sample project but same code i am applying to production app i am getting below error

When i am opening the form for edit that time i am getting below error and when i am closing the error page i can see value of country, state and city are updated/selected no issue.

@eakoriakin  Please help me 

how can i troubleshoot?

Thank you

Error 
```
e.html:10 ERROR TypeError: Cannot read property 'find' of undefined
    at e._formatValueItem (ionic-selectable.min.js:1)
    at Object.eval [as updateRenderer] (e.html:10)
    at Object.debugUpdateRenderer [as updateRenderer] (core.js:14735)
    at checkAndUpdateView (core.js:13849)
    at callViewAction (core.js:14195)
    at execEmbeddedViewsAction (core.js:14153)
    at checkAndUpdateView (core.js:13845)
    at callViewAction (core.js:14195)
    at execEmbeddedViewsAction (core.js:14153)
    at checkAndUpdateView (core.js:13845)
```

Version:
    "ionic-selectable": "^3.2.0",
    "ionic-angular": "3.9.2",

code: 

> <ion-item>
        <ion-label stacked class="label-color">Country</ion-label>
        <ionic-selectable item-content [(ngModel)]="test.country" name="country" [items]="countryList"
            itemValueField="id" itemTextField="name" [canSearch]="true" [shouldStoreItemValue]="true"
            (onChange)="getState()">
            <ng-template ionicSelectableCloseButtonTemplate>
                <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
            </ng-template>
        </ionic-selectable>
    </ion-item>
    <ion-item>
        <ion-label stacked class="label-color">State / District</ion-label>
        <ionic-selectable item-content [(ngModel)]="test.state" name="state" [items]="stateList"
            itemValueField="id" itemTextField="state" [canSearch]="true" [shouldStoreItemValue]="true"
            (onChange)="getCities()">
            <ng-template ionicSelectableCloseButtonTemplate>
                <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
            </ng-template>
        </ionic-selectable>
    </ion-item>

        <ion-item>
            <ion-label stacked class="label-color">City / Upzila</ion-label>
            <ionic-selectable item-content [(ngModel)]="test.city" name="city" [items]="cityList"
                itemValueField="id" itemTextField="name" [canSearch]="true" [shouldStoreItemValue]="true">
                <ng-template ionicSelectableCloseButtonTemplate >
                    <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
                </ng-template>
            </ionic-selectable>
        </ion-item>

even i have tried with only country but still facing the issue

> <ion-item>
        <ion-label stacked class="label-color">Country</ion-label>
        <ionic-selectable item-content [(ngModel)]="test.country" name="country" [items]="countryList"
            itemValueField="id" itemTextField="name" [canSearch]="true" [shouldStoreItemValue]="true"
            (onChange)="getState()">
            <ng-template ionicSelectableCloseButtonTemplate>
                <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
            </ng-template>
        </ionic-selectable>
    </ion-item>

this problem only occur when i add this [shouldStoreItemValue]="true"

**Comment by indraraj26** [comment]
when i click to the error i am pointing to this line

> `e.prototype._formatValueItem=function(e){var t=this;if(this._shouldStoreItemValue){`

**Comment by eakoriakin** [comment]
Hi, can show me your countryList array?

**Comment by indraraj26** [comment]
Dear Eakoriakin,

Here is the list of countryList array:
copy this url and paste in your browser
[https://pastecode.xyz/view/430d7597](url)

here is the output of person that i am trying to edit.
copy this url and paste in your browser
[https://pastecode.xyz/view/1cb9cd20](url)

Thank you.

**Comment by eakoriakin** [comment]
Please, provide an error on StackBlitz with the error, as I don't have time to figure what's wrong with that code sample as I'm working on this project for free.

**Comment by indraraj26** [comment]
I put my all efforts to reproduce the issue but i am failed with sample project. I have used setTimeout to make it like async nature look like prod code without backend. i have no clue where i am doing wrong.

Will you get time to see my prod code with team-viewer only one time please.

Here is the sample project that i have created to reproduce : https://github.com/indraraj26/ionic-selectable-issue

Thank you.

**Comment by indraraj26** [comment]
I will re-open this issue if i get chance to reproduce the issue.

**Comment by crushingCodes** [comment]
I had this issue too. Maybe due to using [(ngModel)]="post.id".
Maybe not the best, but as a workaround for any people having this in the future, I changed it to (ngModel)="post.id". and then changed the onChange event to manually add the desired child property. 
e.g.
```
 categoryChange(event: {
    component: IonicSelectableComponent,
    value: any
  }, type: string) {
    console.log('port:', event.value);
    if (type == "category") {
      this.post.category = event.value.categoryValue
}
}
```

**Comment by ChelloApps** [comment]
I had this error and managed to narrow it down, in my example my selectable was bound using formControlName and I got the reported error if the collection my selectable was bound to was not initialized early enough in the hosting components life cycle.

I should also add that the error only occurred when I was debugging my app on a mobile device, whilst being run in the browser no error was raised.

So in short I prevented the error by defining my collection properties like this: -

public myCollection: Array<ListItem>;

then in the constructor I simply added: -

this.myCollection = [];

This prevented the error from being shown.

Hopefully this might help.

**Comment by petebr** [comment]
same fix here, simply initializing the lookup list as empty array fixed the issue. 

`  myLookupValues : any = []; `

**Comment by surendra-17** [comment]
> I dont get any error in sample project but same code i am applying to production app i am getting below error
> 
> When i am opening the form for edit that time i am getting below error and when i am closing the error page i can see value of country, state and city are updated/selected no issue.
> 
> @eakoriakin Please help me
> 
> how can i troubleshoot?
> 
> Thank you
> 
> Error
> 
> ```
> e.html:10 ERROR TypeError: Cannot read property 'find' of undefined
>     at e._formatValueItem (ionic-selectable.min.js:1)
>     at Object.eval [as updateRenderer] (e.html:10)
>     at Object.debugUpdateRenderer [as updateRenderer] (core.js:14735)
>     at checkAndUpdateView (core.js:13849)
>     at callViewAction (core.js:14195)
>     at execEmbeddedViewsAction (core.js:14153)
>     at checkAndUpdateView (core.js:13845)
>     at callViewAction (core.js:14195)
>     at execEmbeddedViewsAction (core.js:14153)
>     at checkAndUpdateView (core.js:13845)
> ```
> 
> Version:
> "ionic-selectable": "^3.2.0",
> "ionic-angular": "3.9.2",
> 
> code:
> 
> > 
> 
> ```
>     <ion-label stacked class="label-color">Country</ion-label>
>     <ionic-selectable item-content [(ngModel)]="test.country" name="country" [items]="countryList"
>         itemValueField="id" itemTextField="name" [canSearch]="true" [shouldStoreItemValue]="true"
>         (onChange)="getState()">
>         <ng-template ionicSelectableCloseButtonTemplate>
>             <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
>         </ng-template>
>     </ionic-selectable>
> </ion-item>
> <ion-item>
>     <ion-label stacked class="label-color">State / District</ion-label>
>     <ionic-selectable item-content [(ngModel)]="test.state" name="state" [items]="stateList"
>         itemValueField="id" itemTextField="state" [canSearch]="true" [shouldStoreItemValue]="true"
>         (onChange)="getCities()">
>         <ng-template ionicSelectableCloseButtonTemplate>
>             <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
>         </ng-template>
>     </ionic-selectable>
> </ion-item>
> 
>     <ion-item>
>         <ion-label stacked class="label-color">City / Upzila</ion-label>
>         <ionic-selectable item-content [(ngModel)]="test.city" name="city" [items]="cityList"
>             itemValueField="id" itemTextField="name" [canSearch]="true" [shouldStoreItemValue]="true">
>             <ng-template ionicSelectableCloseButtonTemplate >
>                 <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
>             </ng-template>
>         </ionic-selectable>
>     </ion-item>
> ```
> 
> even i have tried with only country but still facing the issue
> 
> > 
> 
> ```
>     <ion-label stacked class="label-color">Country</ion-label>
>     <ionic-selectable item-content [(ngModel)]="test.country" name="country" [items]="countryList"
>         itemValueField="id" itemTextField="name" [canSearch]="true" [shouldStoreItemValue]="true"
>         (onChange)="getState()">
>         <ng-template ionicSelectableCloseButtonTemplate>
>             <ion-icon name="close-circle" style="font-size: 24px;" color="primary"></ion-icon>
>         </ng-template>
>     </ionic-selectable>
> </ion-item>
> ```
> 
> this problem only occur when i add this [shouldStoreItemValue]="true"

initialize your items property variable to empty array like (public countryList:any = []) solve the problem for me
